wallet/src/Pos/Wallet/Web/Methods/History.hs

```diff
@@ -166,22 +169,23 @@ getHistoryLimited mCWalId mAccId mAddrId mSkip mLimit = do
         "Please do not specify both walletId and accountId at the same time"
 
 addHistoryTx
-    :: MonadWalletWebMode m
-    => CId Wal
+    :: (WalletDbWriter A.AddOnlyNewTxMetas m, MonadIO m)
```

Comment:
Are you sure `WalletDbWriter` is needed? We're not abstracting over the event here.